src/lib/bootloader/serial_console.rb

```diff
@@ -0,0 +1,101 @@
+require "yast"
+
+Yast.import "Arch"
+
+module Bootloader
+  # Represents parameters for console. Its main intention is easy parsing serial
+  # console parameters parameters for grub or kernel and generate it to keep it
+  # in sync.
+  class SerialConsole
+    PARITY_MAP = {
+      "n" => "no",
+      "o" => "odd",
+      "e" => "even"
+    }
+    SPEED_DEFAULT = 9600
+    PARITY_DEFAULT = "no"
+    WORD_DEFAULT = ""
+
+    # REGEXP that separate usefull parts of kernel parameter for serial console
+    # matching groups are:
+    #
+    # 1. serial console device
+    # 2. console unit
+    # 3. speed of serial console ( baud rate )
+    # 4. parity of serial console ( just first letter )
+    # 5. word length for serial console
+    #
+    # For details see https://en.wikipedia.org/wiki/Serial_port
+    # @example serial console param ( on kernel cmdline "console=<example>" )
+    #    "ttyS0,9600n8"
+    # @example also partial specification works
+    #    "ttyAMA1"
+    KERNEL_PARAM_REGEXP = /(ttyS|ttyAMA)([[:digit:]]*),?([[:digit:]]*)([noe]*)([[:digit:]]*)/
+
+    # Loads serial console configuration from parameters passed to kernel
+    # @param [ConfigFiles::Grub2::Default::KernelParams] kernel_params to read
```

Comment:
I think it is a needless dependency on the KernelParams class. The other method simply gets a string, so I would expect this one to get a string too.